src/field-builder.ts

```diff
@@ -198,3 +219,26 @@ fieldBuilderProto.effectConnection = function connection(
 export function capitalize(s: string) {
   return `${s.slice(0, 1).toUpperCase()}${s.slice(1)}`;
 }
+
+fieldBuilderProto.prismaEffect = function prismaEffect({ effect = {}, resolve, ...options }) {
+  return this.prismaField({
+    ...options,
+    // FIXME: resolve type error
+    // @ts-expect-error
+    resolve: (_query, parent, args, context, info) => {
+      const query = queryFromInfo({
+        context,
```

Comment:
If you are using t.prismaField you already have the query object, is there something you are working around here?

Reply:
No I didn't. It came from my misunderstood about t.prismaField's query argument.